model/src/main/kotlin/utils/FileArchiver.kt

```diff
@@ -144,6 +144,14 @@ private fun getArchivePath(provenance: Provenance): String =
 private val SHA1_DIGEST by lazy { MessageDigest.getInstance("SHA-1") }
 
 /**
- * Calculate the SHA-1 hash of the string representation of this [Provenance] instance.
+ * Calculate the SHA-1 hash of the storage key of this [Provenance] instance.
  */
-private fun Provenance.hash(): String = SHA1_DIGEST.digest(toString().toByteArray()).toHexString()
+private fun Provenance.hash(): String {
+    val key = vcsInfo?.let {
+        "${it.type}${it.url}${it.resolvedRevision}"
```

Comment:
Even if not necessary, I wonder whether we should insert some separator chars here, like `-` or `@`, just for human readability of this code.

Reply:
I'd prefer not iterating again as the benefit IMO is rather low, and it requires me to do quite a bit of re-work as I've already tested things a bit and am working on the data migration.

Reply:
> and am working on the data migration.

IMO that's something that you should postpone until this PR is approved, to avoid any duplicate work.

Reply:
My preference stays, anyhow - shall I change it?

Reply:
To me readability of the code is fine, `${}` already serves as a separator.

Reply:
Ok, then let's keep it as-is.